src/libcpu/src/interpreter/interpreter_pairedsingle.cpp

```diff
@@ -559,7 +559,12 @@ ps_res(cpu::Core *state, Instruction instr)
       state->fpr[instr.frD].paired1 = extend_float(d1);
    }
 
+   // ps_res never sets FPSCR[XX], so avoid setting it here.
+   const int xx = state->fpscr.xx;
    updateFPSCR(state, oldFPSCR);
+   if (!xx) {
```

Comment:
I actually said exactly that in the comment above. (: Since updateFPSCR() can only set XX, not clear it, the code is correct as written. But I agree your way is more concise (and possibly faster on average), so I'll update it when I'm back at the code, or feel free to make the change yourself when merging.